pkg/config/type.go

```diff
@@ -155,6 +155,7 @@ type ResourceConfig struct {
 
 // WarmPoolConfig is the configuration of Warm Pool of a resource
 type WarmPoolConfig struct {
+	// TODO: Deprecate DesiredSize in favour of using WarmIPTarget since historically they served the same purpose
```

Comment:
Is that possible to handle the conversion automatically instead of exposing two vraiables for same purpose?

Reply:
Yes, I think its possible to remove the variable completely actually. I actually did attempt to do this and refactor during my change but I realised the scope of the change would expand. The reason is that `DesiredSize` is used in multiple places throughout the codebase, and fully removing it would require refactoring across several files and tests. This increased scope will increase the complexity of the current change, which could in turn raise the risk of introducing unintended bugs or regressions. To expedite the change I propose making this a technical debt item rather than addressing it in this pull request.

Reply:
Resolving this after speaking to Hao offline. The consensus is that regardless of logs being potentially useful for troubleshooting, 
- if the log is deemed critical use log.info(...)
- if the log is only useful for debugging, it should be removed